uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveManager.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * This service contains some methods to deal with who is the manager of Leave
+ * Requests. Examples, checking if a contact is the "Leave Approver" of another
+ * contact of if a contact has permission to administer things under Leave and
+ * Absences.
+ */
+class CRM_HRLeaveAndAbsences_Service_LeaveManager {
+
+  /**
+   * Checks if the current logged in user is the Leave Manager for the Contact
+   * with the given ID.
+   *
+   * In order to do this, we check if there's a Relationship between both
+   * contacts, where contact A ($contactID) "has Leave Approved By" contact B
+   * (the logged in), and that if the Relationship is active on the current date.
+   *
+   * Note about implementation: Due to the impossibility of making complex
+   * queries using the CiviCRM API (r.start_date IS NULL OR r.start_date <=
+   * CURDATE()), this method uses a SQL query to check the relationship.
+   *
+   * @param int $contactID
+   *
+   * @return bool
+   */
+  public function currentUserIsLeaveManagerOf($contactID) {
+    $currentUserID = CRM_Core_Session::getLoggedInContactID();
+
+    $relationshipTable = CRM_Contact_BAO_Relationship::getTableName();
+    $relationshipTypeTable = CRM_Contact_BAO_RelationshipType::getTableName();
+
+    $query = "SELECT r.id 
+                FROM {$relationshipTable} r
+                INNER JOIN {$relationshipTypeTable} rt ON rt.id = r.relationship_type_id
+              WHERE r.is_active = 1 AND 
+                    rt.is_active = 1 AND 
+                    rt.name_a_b = 'has Leave Approved By' AND
+                    rt.name_b_a = 'is Leave Approver of' AND
+                    r.contact_id_a = %1 AND
+                    r.contact_id_b = %2 AND
+                    (r.start_date IS NULL OR r.start_date <= %3) AND 
+                    (r.end_date IS NULL OR r.end_date >= %3)
+              ";
+
+    $params = [
+      1 => [$contactID, 'Integer'],
+      2 => [$currentUserID, 'Integer'],
+      3 => [CRM_Utils_Date::processDate('today'), 'String']
```

Comment:
I think, since "today" is explicit, that `CRM_Utils_Date::processDate('today')` is a bit more expressive than `date('Y-m-d')` and it also has the advantage of returning the date in the format expected by Civi (though the format won't really make a difference for this specific case).

Do you still think it should be replaced by date()?